ballsdex/packages/config/components.py

```diff
@@ -29,40 +39,46 @@ def __init__(self, interaction: discord.Interaction, channel: discord.TextChanne
             )
         )
 
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if interaction.user.id != self.new_player.id:
+            await interaction.response.send_message(
+                "You are not allowed to interact with this menu.", ephemeral=True
+            )
+            return False
+        return True
+
     @button(
         label="Accept",
         style=discord.ButtonStyle.success,
         emoji="\N{HEAVY CHECK MARK}\N{VARIATION SELECTOR-16}",
     )
-    async def accept_button(self, interaction: discord.Interaction, item: discord.ui.Button):
+    async def accept_button(self, interaction: discord.Interaction, button: discord.ui.Button):
         config, created = await GuildConfig.get_or_create(guild_id=interaction.guild_id)
         config.spawn_channel = self.channel.id  # type: ignore
         await config.save()
         interaction.client.dispatch(
             "ballsdex_settings_change", interaction.guild, channel=self.channel
         )
         self.stop()
+        if self.message:
+            button.disabled = True
+            try:
+                await self.message.edit(view=self)
+            except discord.HTTPException:
+                pass
         await interaction.response.send_message(
             f"The new spawn channel was successfully set to {self.channel.mention}.\n"
             f"{settings.collectible_name.title()}s will start spawning as"
             " users talk unless the bot is disabled."
         )
 
-        self.accept_button.disabled = True
-        try:
-            await self.original_interaction.followup.edit_message(
-                "@original", view=self  # type: ignore
-            )
-        except discord.HTTPException:
-            pass
-
     async def on_timeout(self) -> None:
         self.stop()
-        for item in self.children:
-            item.disabled = True  # type: ignore
-        try:
-            await self.original_interaction.followup.edit_message(
-                "@original", view=self  # type: ignore
-            )
-        except discord.HTTPException:
-            pass
+        if self.message:
+            for item in self.children:
+                if isinstance(item, discord.ui.Button):
+                    item.disabled = True
+            try:
+                await self.message.edit(view=self)
+            except discord.HTTPException:
+                pass
```

Comment:
the `self.stop()` is not really needed
```suggestion
    async def on_timeout(self) -> None:
        if self.message:
            for item in self.children:
                item: discord.ui.Button
                item.disabled = True
            try:
                await self.message.edit(view=self)
            except discord.HTTPException:
                pass
```